packages/coverage-v8/src/provider.ts

```diff
@@ -497,27 +486,49 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
 
   private async convertCoverage(
     coverage: RawCoverage,
-    projectName?: ProjectName,
-    transformMode?: 'web' | 'ssr',
+    project: WorkspaceProject = this.ctx.getCoreWorkspaceProject(),
+    transformMode?: keyof CoverageFilesByTransformMode,
   ): Promise<CoverageMap> {
-    const viteNode
-      = this.ctx.projects.find(project => project.getName() === projectName)
-        ?.vitenode || this.ctx.vitenode
-    const fetchCache = transformMode
-      ? viteNode.fetchCaches[transformMode]
-      : viteNode.fetchCache
+    let fetchCache = project.vitenode.fetchCache
+
+    if (transformMode) {
+      fetchCache = transformMode === 'browser' ? new Map() : project.vitenode.fetchCaches[transformMode]
+    }
+
     const transformResults = normalizeTransformResults(fetchCache)
 
-    const scriptCoverages = coverage.result.filter(result =>
-      this.testExclude.shouldInstrument(fileURLToPath(result.url)),
-    )
+    async function onTransform(filepath: string) {
+      if (transformMode === 'browser' && project.browser) {
+        const result = await project.browser.vite.transformRequest(filepath.replace(project.config.root, ''))
```

Comment:
Can we have a utility with something like this? (This is from Vite source)

```ts
if (filepath.startsWith(withTrailingSlash(root))) {
    // in root: infer short absolute path from root
    url = filepath.slice(root.length)
  }
```

Reply:
`removeRoot` utility added and used in all cases where `.replace` was.